### PR TITLE
added docker

### PR DESCRIPTION
**Pull request by ArtemReizer**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,7 @@
+MYSQL_DATABASE = book_store
+MYSQL_DATABASE_USERNAME = root
+MYSQL_DATABASE_PASSWORD = qwerty123456
+SPRING_LOCAL_PORT = 8088
+SPRING_DOCKER_PORT = 8080
+MYSQL_PORT = 3307
+DOCKER_MYSQL_PORT = 3306
```
Comment (OleksandrSaven) [inline .env:7]: don't forget about empty line

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+RUN java -Djarmode=layertools -jar application.jar extract
+
+FROM openjdk:17-jdk-alpine
+WORKDIR application
+COPY --from=builder application/dependencies/ ./
+COPY --from=builder application/spring-boot-loader/ ./
+COPY --from=builder application/snapshot-dependencies/ ./
+COPY --from=builder application/application/ ./
+ENTRYPOINT ["java", "org.springframework.boot.loader.JarLauncher"]
+EXPOSE 8080
```
Comment (OleksandrSaven) [inline Dockerfile:14]: same

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+version: "3.8"
+
+services:
+  db:
```
Comment (OleksandrSaven) [inline docker-compose.yaml:4]: it is strange because you use msqldb

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    environment:
+      SPRING_APPLICATION_JSON: '{
+        "spring.datasource.url" : "jdbc:mysql://db:$DOCKER_MYSQL_PORT/$MYSQL_PORT",
+        "spring.datasource.username" : "MYSQL_DATABASE_USERNAME",
+        "spring.datasource.password" : "MYSQL_DATABASE_PASSWORD",
+        "spring.jpa.properties.hibernate.dialect" : "org.hibernate.dialect.MySQLDialect",
+        "spring.datasource.driver-class-name" : "com.mysql.cj.jdbc.Driver",
+        "spring.jpa.hibernate.ddl-auto" : "validate"
+      }'
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
```
Comment (OleksandrSaven) [inline docker-compose.yaml:29]: empty line

**Review by OleksandrSaven — APPROVED** [review]
GJ, see my coments

File: docker-compose.yaml (hunk 3 above)
Comment (vovasalyha) [reply docker-compose.yaml:4]: @OleksandrSaven you can name your service by any name you want, it doesn't matter actually, so it's fine

File: Dockerfile (hunk 2 above)
Comment (vovasalyha) [reply Dockerfile:14]: not resolved

File: docker-compose.yaml (hunk 4 above)
Comment (vovasalyha) [reply docker-compose.yaml:29]: not resolved

**Review by vovasalyha — APPROVED** [review]
GJ, but please add empty lines

**Pull request merged**